bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisBoxHandler.java

```diff
@@ -0,0 +1,371 @@
+                if (command instanceof OnOffType) {
+                    value = ((OnOffType) command == OnOffType.ON) ? "ON" : "OFF";
+                }
+                break;
+            case CHANNEL_TYPE_TARGETTEMP:
+                if (command instanceof QuantityType) {
+                    QuantityType<?> celsiusTemperature = (QuantityType<?>) command;
+                    celsiusTemperature = celsiusTemperature.toUnit(SIUnits.CELSIUS);
+                    if (celsiusTemperature != null) {
+                        double doubleValue = celsiusTemperature.doubleValue();
+                        if (doubleValue < minTemp) {
+                            doubleValue = minTemp;
+                        } else if (doubleValue > maxTemp) {
+                            doubleValue = maxTemp;
+                        }
+                        value = String.valueOf((int) Math.round(doubleValue * 10));
+                    }
+                }
+                break;
+            case CHANNEL_TYPE_MODE:
+            case CHANNEL_TYPE_FANSPEED:
+            case CHANNEL_TYPE_VANESUD:
+            case CHANNEL_TYPE_VANESLR:
+                value = command.toString();
+                break;
+        }
+        if (!value.isEmpty()) {
+            String function = IntesisBoxFunctionEnum.valueOf(channelUID.getId()).getFunction();
+            if (api != null) {
+                logger.trace("Sending command {} to function {}", value, function);
+                api.sendCommand(function, value);
+            } else {
+                logger.trace("Sending command failed, could not get API");
+            }
+        }
+    }
+
+    private void receivedId(String function, String value) {
+        logger.trace("receivedID(): {} {}", function, value);
+        properties.put(PROPERTY_VENDOR, "Intesis");
+        switch (function) {
+            case "MODEL":
+                properties.put(PROPERTY_MODEL_ID, value);
+                break;
+            case "MAC":
+                properties.put(PROPERTY_MAC_ADDRESS, value);
+                break;
+            case "IP":
+                properties.put("ipAddress", value);
+            case "PROTOCOL":
+                properties.put("protocol", value);
+                break;
+            case "VERSION":
+                properties.put(PROPERTY_FIRMWARE_VERSION, value);
+                break;
+            case "RSSI":
+                properties.put("rssi", value);
+                break;
+            case "NAME":
+                properties.put("hostname", value);
+                break;
+        }
+    }
+
+    private void receivedUpdate(String function, String value) {
+        logger.trace("receivedUpdate(): {} {}", function, value);
+        switch (function) {
+            case "ONOFF":
+                updateState(CHANNEL_TYPE_POWER, OnOffType.from(value));
+                break;
+
+            case "SETPTEMP":
+                if (value.equals("32768")) {
+                    value = "0";
```

Comment:
It's bad practice to modify method arguments. That's why the compiler complains. Better create a new local variable for it.

Reply:
Done